src/gromacs/nblib/topology.cpp

```diff
@@ -140,6 +157,105 @@ void ParticleSequencer::build(const std::vector<std::tuple<Molecule, int>>& mole
     }
 }
 
+template<class B>
+std::vector<B> aggregateBonds(const std::vector<std::tuple<Molecule, int>>& molecules)
```

Comment:
This seems rather more like it is unpacking or expanding bonds. I would suggest a much more explicit name like `expandBondsToMatchMoleculeNumber`.

Reply:
If not clear from the function signature, "ToMatchMoleculeNumber" can be clarified with a quick bit of documentation. The purpose of this function is to get a list of bonds of a particular type from a list of molecules? `makeTypedBondList` might be a more appropriate name, or `listBondsByType` if you add a parameter to allow template type deduction of B instead of requiring the template parameter.

Reply:
I'm proposing `collectBonds`. The doc strings says "extract all bonds from a vector" to which I'm going to add "the second tuple element specifies multiples of the molecule given as first tuple element". This makes it clear that it collects all the bonds from the input molecules and puts them in one vector.